designs/tls/install-cert-manager.md

```diff
@@ -0,0 +1,187 @@
+---
+oep-number: OEP 3855
+title: Enable TLS in Mayastor
+authors:
+  - "@Johnaius"
+owners:
+  - "@Johnaius"
+editor: TBD
+creation-date: 02/6/2025
+last-updated: 02/6/2025
+status: implementable
+---
+
+# Enable TLS in Mayastor
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+    * [Story 1](#story-1)
+    * [Story 2](#story-2)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    * [Discoverable images](#discoverable-images)
+      * [Regex Explanation](#regex-explanation)
+    * [Expose Images](#expose-images)
+    * [Non-Discoverable/Runtime images](#non-discoverableruntime-images)
+  * [Test Plan](#test-plan)
+  * [Risks and Mitigations](#risks-and-mitigations)
+    * [Mitigations](#mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+This proposal aims to add TLS communication between the components of Mayastor
+
+## Motivation
+
+Security
+
+### Goals
+
+1. Create TLS switch
+
+1. Install Cert-Manager
+
+1. Generate a Self signed certs
+
+1. Ensure each component is configured to have appropriate certs if tls is enabled. See diagram below.
+    * will need help identifying how to secure comms with etcd.
+
+1. Update Mayastor components to use the generated certificates
+![Diagram](./images/flowofmayastor.png)
+
+## Proposal
+
+The proposal is to add cert-manager to mayastor to automate the creation of certificates.  Also, values.yaml file will be updated to include an enable tls switch boolean.  Certificates will be created with mayastor certificate resources. This will include a CA certificate which will be used to sign the certificates for rest-api, core-agent and io-engine.  These certificates will be mounted into each deployment/daemonset and distributed as well to each client.
+
+### User Stories
+
+#### Story 1
+
+As a user, I want to ensure that my data is securely transmitted between Mayastor components so that I can trust the integrity and confidentiality of my data during storage operations.
+
+#### Story 2
+
+As a user, I want to not have to worry about certificate creation or management.
+
+### Implementation Details/Notes/Constraints
+
+#### Cert Manager
+
+Cert-manager installation shall be done using [helm charts][certManagerHelmCharts]
+
+* Should there be discussion about drawbacks of depending on cert-manager or other strategies?
```

Comment:
Any suggestions on what other strategies would be?
We can highlight them on an Alternatives section: https://github.com/openebs/openebs/blob/develop/designs/oep-template.md#alternatives-optional

Reply:
Manually created is the only other thing I thought of. I wasn't sure if you wanted to depend on cert-manager. Does more thought need to go into that?

Reply:
I think depending on cert-manager is the way.

Reply:
I think cert-manager is the way. the only drawback I can think if is a slightly longer install time.  It appears cert-manager-webhook container takes roughly 90s to be ready to go.